src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Guid.txt

```diff
@@ -0,0 +1,79 @@
+﻿// Guid function 
+// https://docs.microsoft.com/en-us/powerapps/maker/canvas-apps/functions/function-guid
+
+// Guid with no args is volatile, so we can't directly test the output
+
+>> IsBlank(GUID())
+false
+
+>> IsError(GUID())
+false
+
+// Multiple invocations are unique
+>> GUID() <> GUID()
+true
+
+// Coercion to string is 32 chars + 4 hyphens
+>> Len(GUID())
+36
+
+>> GUID(Blank())
+Blank()
+
+>> GUID(Text(1/0))
+#Error(Kind=Div0)
+
+>> GUID("c3987aa4-52ff-462f-9a7c-a13c0dedb16d")
+GUID("c3987aa4-52ff-462f-9a7c-a13c0dedb16d")
+
+>> GUID("c3987aa4-52ff-462f-9a7c-a13c0dedb16d") & "aaaaaaaaaaaaaaaa"
+"c3987aa4-52ff-462f-9a7c-a13c0dedb16daaaaaaaaaaaaaaaa"
+
+>> GUID("8A9E8006-3EDE-4584-863F-4226BAA73D08")
+GUID("8a9e8006-3ede-4584-863f-4226baa73d08")
+
+>> GUID("0A6D72E27FDE47C5BB9513BDB51BD7C0")
+GUID("0a6d72e2-7fde-47c5-bb95-13bdb51bd7c0")
+
+>> GUID("0c99c9a834ed4247861dc9680d72dfd9")
+GUID("0c99c9a8-34ed-4247-861d-c9680d72dfd9")
+
+>> Text(GUID("41aacfaa-8630-4d9b-8cdd-f62ed8ee8581"))
+"41aacfaa-8630-4d9b-8cdd-f62ed8ee8581"
+
+>> GUID("76353e29-1d57-4315-8af1-884ce7e6c75c") = GUID("76353e29-1d57-4315-8af1-884ce7e6c75c")
+true
+
+>> GUID("76353e291d5743158af1884ce7e6c75c") = GUID("76353E29-1D57-4315-8AF1-884CE7E6C75C")
+true
+
+// Not comparable to string
+>> GUID("76353e29-1d57-4315-8af1-884ce7e6c75c") = "76353e29-1d57-4315-8af1-884ce7e6c75c"
+Errors: Error 45-46: GUID values can only be compared to other GUID values.
+
+>> GUID("NotAGuid")
+#Error(Kind=InvalidArgument)
+
+// Missing 1 char
+>> GUID("ad4c12ae-e897-4bdc-8e37-8a2bfa04636")
+#Error(Kind=InvalidArgument)
+
+// 1 extra char
+>> GUID("743c9c6fb-a9ab-476e-8231-8c76631fb6f0")
+#Error(Kind=InvalidArgument)
+
+// Not a valid guid
+>> GUID("560fd2fc-cef6-xxxx-ad40-628a9d2cb164")
+#Error(Kind=InvalidArgument)
+
+
+
+// Not currently supported formats
+>> GUID("{275238ab-f73d-4167-862b-7d6c4e0e288c}")
+#Error(Kind=InvalidArgument
```

Comment:
Missing paren?

Reply:
hm. This should have been a failure with that mismatch, looking into why it passed.

Reply:
The regex looks for it is "lazy" - it will only enforce the error kind if it is defined properly, and ignore the kind otherwise. We should be more strict here - either allow `#Error` only, or force it to be properly formed.

Reply:
We could fix the regex issue in a future PR. Just fix the missing paren here and then ship.